nu che pognali

Perevechenkov opened the pull request.

Perevechenkov committed:
looks ez

Perevechenkov committed:
not ez(

Perevechenkov committed:
tessst

Perevechenkov committed:
testi

Perevechenkov committed:
let's add some crutches

Perevechenkov committed:
absolutely nothing

AlexNg9 commented on index.js:
```diff
@@ -6,20 +6,34 @@
 function customBind (func, context, ...args) {
-    // code
+    return function () {
+        const inputArgs = [].slice.call(arguments);
```
Аргументы можно собрать прямо в сигнатуре функции.

AlexNg9 commented on index.js:
```diff
@@ -6,20 +6,34 @@
  * Напишите функцию sum, вычисляющую суммы подобным образом:
  * sum(1)(2)( ) // 3
- * sum(1)(2)(3)( ) // 6
+ * sum(1)(2)(3)(4)( ) // 10
```
Тоже верно.

AlexNg9 commented on index.js:
```diff
@@ -6,20 +6,34 @@
  * sum :: Number -> sum
  * sum :: void -> Number
  */
+let accumulator = 0;
```
То, что переменная доступна вне функции – плохо. К ней можно будет обратиться из других функций. Нужно ее как-то спрятать.

Perevechenkov committed:
fixed bind, changed sum

call с псевдомассивом заменил на rest параметр.
В sum изменил логику работы, используя массив для записи элементов

DesperateHiking commented:
как-то тупо коммит назвал, больше так не буду

AlexNg9 commented on index.js:
```diff
@@ -22,18 +20,22 @@ function customBind (func, context, ...args) {
-        accumulator = 0;
+    if (x === undefined) return 0;
+
+    const accumulator = [];
```
Зачем массив? Почему не через число?

Perevechenkov replied:
сейчас все будет((

Perevechenkov replied:
а вообще просто захотелось что-то нетакое как у всех сделать
![cD8dxkoOYFA](https://user-images.githubusercontent.com/48237694/67554605-2a7bff80-f729-11e9-98b3-4c7feafce44f.jpg)

AlexNg9 replied:
@desperateHiking, окей, понял
Но это избыточно все-таки

Perevechenkov committed:
using array was bad idea

AlexNg9 commented on index.js:
```diff
@@ -23,19 +23,11 @@ function customBind (func, context, ...args) {
-        accumulator.push(nextItem);
-
-        return addNextItem;
+        return sum(x + nextNum);
```
Вот это очень хороший вариант. Даже дополнительную переменную не нужно создавать.
